From 507159be87b2a75ea50054bbcc4e895349fdab52 Mon Sep 17 00:00:00 2001
From: Leo Soares <leojrfs@gmail.com>
Date: Tue, 26 Jul 2016 15:45:31 +0100
Subject: [PATCH] removed ucontext.h deps

---
 cpu/native/mtarch.c | 89 -----------------------------------------------------
 1 file changed, 89 deletions(-)

diff --git a/cpu/native/mtarch.c b/cpu/native/mtarch.c
index da55744..6f596d3 100644
--- a/cpu/native/mtarch.c
+++ b/cpu/native/mtarch.c
@@ -38,20 +38,6 @@
 #define MTARCH_STACKSIZE 4096
 #endif /* MTARCH_STACKSIZE */
 
-#if defined(_WIN32) || defined(__CYGWIN__)
-
-#define WIN32_LEAN_AND_MEAN
-#include <windows.h>
-
-static void *main_fiber;
-
-#elif defined(__linux) || defined(__APPLE__)
-
-#ifdef __APPLE__
-/* Avoid deprecated error on Darwin */
-#define _XOPEN_SOURCE
-#endif
-
 #include <stdlib.h>
 #include <signal.h>
 #include <ucontext.h>
@@ -64,27 +50,16 @@ struct mtarch_t {
 static ucontext_t main_context;
 static ucontext_t *running_context;
 
-#endif /* _WIN32 || __CYGWIN__ || __linux */
 
 /*--------------------------------------------------------------------------*/
 void
 mtarch_init(void)
 {
-#if defined(_WIN32) || defined(__CYGWIN__)
-
-  main_fiber = ConvertThreadToFiber(NULL);
-
-#endif /* _WIN32 || __CYGWIN__ */
 }
 /*--------------------------------------------------------------------------*/
 void
 mtarch_remove(void)
 {
-#if defined(_WIN32) || defined(__CYGWIN__)
-
-  ConvertFiberToThread();
-
-#endif /* _WIN32 || __CYGWIN__ */
 }
 /*--------------------------------------------------------------------------*/
 void
@@ -92,85 +67,21 @@ mtarch_start(struct mtarch_thread *thread,
 	     void (* function)(void *data),
 	     void *data)
 {
-#if defined(_WIN32) || defined(__CYGWIN__)
-
-  thread->mt_thread = CreateFiber(0, (LPFIBER_START_ROUTINE)function, data);
-
-#elif defined(__linux)
-
-  thread->mt_thread = malloc(sizeof(struct mtarch_t));
-
-  getcontext(&((struct mtarch_t *)thread->mt_thread)->context);
-
-  ((struct mtarch_t *)thread->mt_thread)->context.uc_link = NULL;
-  ((struct mtarch_t *)thread->mt_thread)->context.uc_stack.ss_sp = 
-			((struct mtarch_t *)thread->mt_thread)->stack;
-  ((struct mtarch_t *)thread->mt_thread)->context.uc_stack.ss_size = 
-			sizeof(((struct mtarch_t *)thread->mt_thread)->stack);
-
-  /* Some notes:
-     - If a CPU needs stronger alignment for the stack than malloc()
-       guarantees (like i.e. IA64) then makecontext() is supposed to
-       add that alignment internally.
-     - According to POSIX the arguments to function() are of type int
-       and there are in fact 64-bit implementations which support only
-       32 bits per argument meaning that a pointer argument has to be
-       splitted into two arguments.
-     - Most implementations interpret context.uc_stack.ss_sp on entry
-       as the lowest stack address even if the CPU stack actually grows
-       downwards. Although this means that ss_sp does NOT represent the
-       CPU stack pointer this behaviour makes perfectly sense as it is
-       the only way to stay independent from the CPU architecture. But
-       Solaris prior to release 10 interprets ss_sp as highest stack
-       address thus requiring special handling. */
-  makecontext(&((struct mtarch_t *)thread->mt_thread)->context,
-	      (void (*)(void))function, 1, data);
-
-#endif /* _WIN32 || __CYGWIN__ || __linux */
 }
 /*--------------------------------------------------------------------------*/
 void
 mtarch_yield(void)
 {
-#if defined(_WIN32) || defined(__CYGWIN__)
-
-  SwitchToFiber(main_fiber);
-
-#elif defined(__linux)
-
-  swapcontext(running_context, &main_context);
-
-#endif /* _WIN32 || __CYGWIN__ || __linux */
 }
 /*--------------------------------------------------------------------------*/
 void
 mtarch_exec(struct mtarch_thread *thread)
 {
-#if defined(_WIN32) || defined(__CYGWIN__)
-
-  SwitchToFiber(thread->mt_thread);
-
-#elif defined(__linux)
-
-  running_context = &((struct mtarch_t *)thread->mt_thread)->context;
-  swapcontext(&main_context, running_context);
-  running_context = NULL;
-
-#endif /* _WIN32 || __CYGWIN__ || __linux */
 }
 /*--------------------------------------------------------------------------*/
 void
 mtarch_stop(struct mtarch_thread *thread)
 {
-#if defined(_WIN32) || defined(__CYGWIN__)
-
-  DeleteFiber(thread->mt_thread);
-
-#elif defined(linux) || defined(__linux)
-
-  free(thread->mt_thread);
-
-#endif /* _WIN32 || __CYGWIN__ || __linux */
 }
 /*--------------------------------------------------------------------------*/
 void
